test-suite/shared/src/test/scala/org/scalajs/testsuite/niobuffer/ByteBufferTest.scala

```diff
@@ -129,7 +129,7 @@ abstract class ByteBufferTest extends BaseBufferTest {
 
     buf.order(ByteOrder.BIG_ENDIAN)
     val charBuf1 = buf.asCharBuffer()
-    assertEquals(createsReadOnly, charBuf1.isReadOnly)
+    assertEquals(createsReadOnly, charBuf1.asInstanceOf[java.nio.Buffer].isReadOnly)
```

Comment:
since this is an upcast, you can use `: java.nio.Buffer`, instead of `asInstanceOf`, the latter is unsafe since it will still compile even if the type of `charBuf1` changes in the future.